Sources/Networking/AlamofireNetworkClient/AlamofireNetworkClient.swift

```diff
@@ -182,8 +218,11 @@ public extension AlamofireNetworkClient.Request {
         switch $0.result {
         case .success(let json):
           promise.resolve(with: json)
+          self.eventMonitors.forEach { $0.requestDidSucceed(self) }
         case .failure(let error):
-          promise.reject(with: self.handleError(error))
+          let error = self.handleError(error)
+          promise.reject(with: error)
+          self.eventMonitors.forEach { $0.requestDidFail(self, with: error) }
```

Comment:
Should we trigger the event before fulfilling promise?

Reply:
Done: https://github.com/poviolabs/PovioKit/pull/118/commits/194e31e736ad734e998294562d2e896b2204a35d